src/hooks/useRequestProcessor.js

```diff
@@ -0,0 +1,66 @@
+import { useCallback } from 'react'
+import { useHeartbeat } from './useCourtContracts'
+import { useCourtClock } from '../providers/CourtClock'
+import { useRequestQueue } from '../providers/RequestQueue'
+import { useActivity } from '../components/Activity/ActivityProvider'
```

Comment:
This is separate from this, but perhaps we should move this component to the `providers`? Maybe you've already had this conversation before haha.

Reply:
Ah yes! I had some doubts on where to place this hook at the beginning haha. Agree that placing it in the `providers` folder makes sense